test/parallel/test-stream-transform-by.js

```diff
@@ -0,0 +1,249 @@
+'use strict';
+const { mustCall } = require('../common');
+const { once } = require('events');
+const { Readable, Transform } = require('stream');
+const { strictEqual } = require('assert');
+
+async function transformBy() {
+  const readable = Readable.from('test');
+  async function * mapper(source) {
+    for await (const chunk of source) {
+      yield chunk.toUpperCase();
+    }
+  }
+
+  const stream = Transform.by(mapper);
+  readable.pipe(stream);
+  const expected = ['T', 'E', 'S', 'T'];
+  for await (const chunk of stream) {
+    strictEqual(chunk, expected.shift());
+  }
+}
+
+async function transformByFuncReturnsObjectWithSymbolAsyncIterator() {
+  const readable = Readable.from('test');
+  const mapper = (source) => ({
+    [Symbol.asyncIterator]() {
+      return {
+        async next() {
+          const { done, value } = await source.next();
+          return { done, value: value ? value.toUpperCase() : value };
+        }
+      };
+    }
+  });
+
+  const stream = Transform.by(mapper);
+  readable.pipe(stream);
+  const expected = ['T', 'E', 'S', 'T'];
+  for await (const chunk of stream) {
+    strictEqual(chunk, expected.shift());
+  }
+}
+
+async function transformByEncoding() {
+  const readable = Readable.from('test');
+  async function * mapper(source) {
+    for await (const chunk of source) {
+      strictEqual(source.encoding, 'ascii');
```

Comment:
Would that fix the test failure?
```suggestion
      strictEqual(chunk.encoding, 'ascii');
```

Reply:
I guess point of the test is to make `encoding` of previous source stream available to the transformer, without transformer having to keep reference in some other ways. If we were to apply this change we might as well just remove the test.